src/hrflow_connectors/connectors/hrflow/warehouse/profile.py

```diff
@@ -202,13 +203,58 @@ def write_parsing(
             metadatas=profile["metadatas"],
             created_at=profile["created_at"],
         )
-        if response["code"] != 202:
+        if parsing_response["code"] not in [202, 201]:  # 202: Accepted, 201: Created
             adapter.error(
                 "Failed to parse profile with reference={} response={}".format(
-                    profile["reference"], response
+                    profile["reference"], parsing_response
                 )
             )
             failed.append(profile)
+        source_response = hrflow_client.source.get(
+            key=parameters.source_key
+        )  # Get source to check if sync_parsing is enabled
+        if source_response["code"] != 200:
+            adapter.warning(
+                "Failed to get source with key={} response={}, won't be able to update"
+                " profile parsed with profile json".format(
+                    parameters.source_key, source_response
+                )
+            )
+            continue
+        elif source_response["data"]["sync_parsing"] is True:
+            current_profile = parsing_response["data"]["profile"]
+            if current_profile:
+                info_parsed = current_profile.get("info", {})
+                if "urls" in profile_info and isinstance(profile_info["urls"], list):
+                    for new_url in profile_info["urls"]:
+                        existing_urls = info_parsed.get("urls", [])
+                        if new_url not in existing_urls:
+                            existing_urls.append(new_url)
+                        info_parsed["urls"] = existing_urls
```

Comment:
Ici quand même le code est compliqué 

```python
for new_url in profile_info["urls"]:
      existing_urls = info_parsed.get("urls", []) <=== Ici pour chaque element `new_url` tu récupères info_parsed.urls
      if new_url not in existing_urls:
            existing_urls.append(new_url).  <=== eventuellement tu rajoutes
      info_parsed["urls"] = existing_urls. <==== tu écrases et remet la même liste
       dans info_parsed.url juste à temps pour la prochaine boucle ou tu récupère 
       encore une fois en début de boucle . Sachant que a priori en plus ca fait rien 
      parce que de base info_parsed["urls"] pointe déjà vers existing_urls 
```

Pourquoi toutes ces complications ?

Pourquoi est ce que tu loops par sur `profile_info["urls"]` puis tu check `if url not in info_parsed["urls"]` alors tu rajoutes directement `info_parsed["urls"].append(url)` ? 

Essaye de réecrire le code pour faire plus simple stp

Reply:
Essaye chez toi en local de faire un truc du genre 
```python
a = dict(b=[1, 2, 3])
my_list = a["b"]
my_list.append(4)

print(a["b"]) # Normalement tu devrais avoir 4 sans avoir a faire avant a["b"] = my_list
```